### PR TITLE
8274656: Remove default_checksum and safe_checksum_type from krb5.conf

### PR DESCRIPTION
**Pull request by wangweij**
Remove support for the 2 settings. Remove classes around the KRB_PRIV and KRB_SAFE message types that were never used in Java.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8274656](https://bugs.openjdk.java.net/browse/JDK-8274656): Remove default_checksum and safe_checksum_type from krb5.conf

### Reviewers
 * [Valerie Peng](https://openjdk.java.net/census#valeriep) (@valeriepeng - **Reviewer**) ⚠️ Review applies to 83b9492000ba043b360b6758a47f0e33a1e5ce36

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5788/head:pull/5788` \
`$ git checkout pull/5788`

Update a local copy of the PR: \
`$ git checkout pull/5788` \
`$ git pull https://git.openjdk.java.net/jdk pull/5788/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5788`

View PR using the GUI difftool: \
`$ git pr show -t 5788`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5788.diff">https://git.openjdk.java.net/jdk/pull/5788.diff</a>

</details>

### CODE REVIEW

File: src/java.security.jgss/share/classes/sun/security/krb5/Checksum.java [hunk 1]
```diff
@@ -73,56 +73,7 @@
     // draft-brezak-win2k-krb-rc4-hmac-04.txt
     public static final int CKSUMTYPE_HMAC_MD5_ARCFOUR = -138;
```
Comment (valeriepeng) [inline src/java.security.jgss/share/classes/sun/security/krb5/Checksum.java:75]: nit: How about updating and fixing the copyright years?

Comment (wangweij) [reply src/java.security.jgss/share/classes/sun/security/krb5/Checksum.java:75]: Will update it in the next commit.

File: src/java.security.jgss/share/classes/sun/security/krb5/KrbTgsReq.java [hunk 2]
```diff
@@ -318,7 +318,7 @@ private TGSReq createRequest(
         byte[] temp = reqBody.asn1Encode(Krb5.KRB_TGS_REQ);
         // if the checksum type is one of the keyed checksum types,
         // use session key.
```
Comment (valeriepeng) [inline src/java.security.jgss/share/classes/sun/security/krb5/KrbTgsReq.java:320]: nit: does this comment need updating?

Comment (wangweij) [reply src/java.security.jgss/share/classes/sun/security/krb5/KrbTgsReq.java:320]: Will remove it. I think there is no need to explain why -1 is used.

**Comment by valeriepeng** [comment]
Do we still need all the constants defined in sun/security/krb5/internal/cryptoKeyUsage.java now that the corresponding classes are removed?

**Comment by wangweij** [comment]
> Do we still need all the constants defined in sun/security/krb5/internal/cryptoKeyUsage.java now that the corresponding classes are removed?

I haven't touched this file, and I also haven't cleaned up constants in `sun/security/krb5/internal/Krb5.java`. I thought they can just be left there as placeholders.

File: src/java.security.jgss/share/classes/sun/security/krb5/KrbTgsReq.java (hunk 2 above)
Comment (valeriepeng) [reply src/java.security.jgss/share/classes/sun/security/krb5/KrbTgsReq.java:320]: Sure

**Review by valeriepeng — APPROVED** [review]
Looks good. Thanks.

**Pull request closed without merging**